sdk/core/azure-core/src/cryptography/md5.cpp

```diff
@@ -16,100 +16,71 @@
 #include <stdexcept>
 #include <vector>
 
-namespace Azure { namespace Core { namespace Cryptography {
+namespace {
 
 #if defined(AZ_PLATFORM_WINDOWS)
 
-  namespace _detail {
-    struct AlgorithmProviderInstance
-    {
-      BCRYPT_ALG_HANDLE Handle;
-      std::size_t ContextSize;
-      std::size_t HashLength;
-
-      AlgorithmProviderInstance()
-      {
-        NTSTATUS status = BCryptOpenAlgorithmProvider(&Handle, BCRYPT_MD5_ALGORITHM, nullptr, 0);
-        if (!BCRYPT_SUCCESS(status))
-        {
-          throw std::runtime_error("BCryptOpenAlgorithmProvider failed");
-        }
-        DWORD objectLength = 0;
-        DWORD dataLength = 0;
-        status = BCryptGetProperty(
-            Handle,
-            BCRYPT_OBJECT_LENGTH,
-            reinterpret_cast<PBYTE>(&objectLength),
-            sizeof(objectLength),
-            &dataLength,
-            0);
-        if (!BCRYPT_SUCCESS(status))
-        {
-          throw std::runtime_error("BCryptGetProperty failed");
-        }
-        ContextSize = objectLength;
-        DWORD hashLength = 0;
-        status = BCryptGetProperty(
-            Handle,
-            BCRYPT_HASH_LENGTH,
-            reinterpret_cast<PBYTE>(&hashLength),
-            sizeof(hashLength),
-            &dataLength,
-            0);
-        if (!BCRYPT_SUCCESS(status))
-        {
-          throw std::runtime_error("BCryptGetProperty failed");
-        }
-        HashLength = hashLength;
-      }
-
-      ~AlgorithmProviderInstance() { BCryptCloseAlgorithmProvider(Handle, 0); }
-    };
-
-    struct Md5HashContext
-    {
-      std::string buffer;
-      BCRYPT_HASH_HANDLE hashHandle = nullptr;
-      std::size_t hashLength = 0;
-    };
-  } // namespace _detail
+struct Md5AlgorithmProvider
```

Comment:
This should be a class per our design patterns.

Reply:
You mean this guideline:
https://azure.github.io/azure-sdk/cpp_implementation.html#cpp-design-naming-classes-no-struct-keyword
???

This class has only public members, it's following the guideline